src/Core/src/Platform/iOS/KeyboardAutoManager.cs

```diff
@@ -0,0 +1,44 @@
+﻿/*
+ * This class is adapted from IQKeyboardManager which is an open-source
+ * library implemented for iOS to handle Keyboard interactions with
+ * UITextFields/UITextViews. Link to their MIT License can be found here:
+ * https://github.com/hackiftekhar/IQKeyboardManager/blob/7399efb730eea084571b45a1a9b36a3a3c54c44f/LICENSE.md
+ */
+
+using System;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManager
+{
+	internal static void GoToNextResponderOrResign(UIView view, bool isUnchangeableReturnKey = false, UIView? customSuperView = null)
+	{
+		if (!view.CheckIfEligible(isUnchangeableReturnKey))
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var superview = customSuperView ?? view.GetViewController<ContainerViewController>()?.View;
+		if (superview is null)
+		{
+			view.ResignFirstResponder();
+			return;
+		}
+
+		var nextField = view.FindNextView(superview, new Type[] { typeof(UITextView), typeof(UITextField) });
+		view.ChangeFocusedView(nextField);
+	}
+
+	static bool CheckIfEligible(this UIView view, bool isUnchangeableReturnKey)
+	{
+		// have isUnchangeableReturnKey flag since EntryCells do not have a public property to change ReturnKeyType
```

Comment:
I don't think we need to special case the `EntryCells` here. If at some point we add `ReturnType` or if the user changes the return type themselves then it'll just opt in with the normal flow.